### PR TITLE
restrict limit of requests how many can robot ask from eshop

### PR DESCRIPTION
**Pull request by sspooky13**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Hello, I recommend you restrict robots on how many requests can ask from eshop. These changes are not officially supported but some robots accept them...it's better as nothing, right? 😅 Thanks 🤝
|New feature| Yes
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: project-base/app/src/DataFixtures/Demo/SettingValueDataFixture.php [hunk 1]
```diff
@@ -117,7 +117,7 @@ public function load(ObjectManager $manager)
             );
             $this->setting->setForDomain(
                 SeoSettingFacade::SEO_ROBOTS_TXT_CONTENT,
-                'Disallow: *?filter=',
+                "Disallow: *?filter=\n\nUser-agent: *\nCrawl-delay: 3\nRequest-rate: 300/1m",
```
Comment (TomasLudvik) [inline project-base/app/src/DataFixtures/Demo/SettingValueDataFixture.php:120]: This should be mentioned in UPGRADE notes (see #project-base-diff...)

File: packages/framework/src/Migrations/Version20230919173422.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            if (count(preg_grep('/User-agent: \*/', $seoRobotsTxtLines)) !== 0) {
+                continue;
+            }
+
+            $seoRobotsTxtContent .= "\n\nUser-agent: *\nCrawl-delay: 3\nRequest-rate: 300/1m";
+            $this->sql('UPDATE setting_values SET value = :value where name = \'seoRobotsTxtContent\'  AND domain_id = :domainId', [
+                'value' => $seoRobotsTxtContent,
+                'domainId' => $domainId,
+            ]);
+        }
```
Comment (TomasLudvik) [inline packages/framework/src/Migrations/Version20230919173422.php:34]: You should mention this in upgrade notes like this:

this change adds migration Version20230919173422 that adds SEO robot limits to your `robots.txt` file. If you do not want to change it in your project, add it as skipped migration to migrations-lock.yaml

**Review by TomasLudvik — CHANGES_REQUESTED** [review]

**Comment by TomasLudvik** [comment]
Also, rebase this branch on the current 15.0 so it does not include commits not associated with this change

File: project-base/app/src/DataFixtures/Demo/SettingValueDataFixture.php (hunk 1 above)
Comment (sspooky13) [reply project-base/app/src/DataFixtures/Demo/SettingValueDataFixture.php:120]: you are right, fixed

File: packages/framework/src/Migrations/Version20230919173422.php (hunk 2 above)
Comment (sspooky13) [reply packages/framework/src/Migrations/Version20230919173422.php:34]: hmm, I hope I fixed it correctly, check it one more time please 🙏

File: UPGRADE-15.0.md [hunk 3]
```diff
@@ -276,6 +276,17 @@ Follow the instructions in relevant sections, e.g. `shopsys/coding-standards` or
...
 -   see #project-base-diff to update your project
 
+#### restrict limit of requests how many can robot ask from eshop ([#2820](https://github.com/shopsys/shopsys/pull/2820))
+
+-   this change adds migration `Version20230919173422` that adds SEO robot limits to your `robots.txt` file. If you do not want to change it in your project, add it as skipped migration to `migrations-lock.yaml`
+-   if you have defined rules for all robots (`User-agent: *`), then you have to manually add these lines to your robots.txt
+    ```diff
+            User-agent: *
+        +   Crawl-delay: 3
+        +   Request-rate: 300/1m
```
Comment (TomasLudvik) [inline UPGRADE-15.0.md]:
```suggestion
    ```diff
        User-agent: *
    +   Crawl-delay: 3
    +   Request-rate: 300/1m
```

**Review by TomasLudvik — APPROVED** [review]

**Review by pk16011990 — APPROVED** [review]

**Pull request merged**